src/audio/eq_fir.c

```diff
@@ -688,10 +672,19 @@ static int eq_fir_copy(struct comp_dev *dev)
 	}
 
 	/* Run EQ function */
-	if (cl.frames & 1)
-		cd->eq_fir_func(fir, cl.source, cl.sink, cl.frames, nch);
-	else
+	if (cl.frames > 1) {
+		/* Decrease number of frames to process if it is odd. The
+		 * optimized FIR function to process even number of frames
+		 * is lower load than generic version.
+		 */
+		if (cl.frames & 1)
+			cl.frames--;
```

Comment:
Sorry, here's a mistake. I need to decrement also the amount of bytes for buffers update.

Reply:
Also good practice to use 0x hex values when doing logical operations.

Reply:
OK, I'll change that.